waf.go

```diff
@@ -77,6 +78,9 @@ func NewWAF(config WAFConfig) (WAF, error) {
 	}
 
 	if r := c.requestBody; r != nil {
+		if r.limit < r.inMemoryLimit {
+			return nil, errors.New("request body limit should be at least the memory limit")
```

Comment:
This is the important change as `inMemoryLimit` tells how much we buffer and `limit` tells how much we read and it does not make sense, limit > inMemoryLimit.

Reply:
Can you add a test?